azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -0,0 +1,357 @@
+            subscription_id=self.default_subscription_id,
+            keyvault_name=spc_keyvault_name,
+        )
+
+        secrets: PageIterator = self.keyvault_client.list_properties_of_secrets()
+
+        # get file extension
+        file_name = os.path.basename(file)
+        # get cert name by removing extension and path in front
+        cert_name = file_name.split(".")[0].replace(".", "")
+
+        cert_extension = validate_file_extension(file_name, ["der", "crt", "crl"])
+
+        try:
+            opcua_secret_sync = self.ssc_mgmt_client.secret_syncs.get(
+                resource_group_name=resource_group,
+                secret_sync_name=OPCUA_ISSUER_LIST_SECRET_SYNC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_secret_sync = {}
+
+        if cert_extension == "crl":
+            # TODO: add check that same name should exist for crt and der
+            secret_mapping = opcua_secret_sync.get("properties", {}).get("objectSecretMapping", [])
+            possible_file_name = [f"{cert_name}.crt", f"{cert_name}.der"]
+            found_file_name = [
+                mapping["targetKey"] for mapping in secret_mapping if mapping["targetKey"] in possible_file_name
+            ]
+
+            if not found_file_name:
+                logger.error(f"Cannot add .crl {file_name} without corresponding .crt or .der file.")
+                return
+
+        secret_name = secret_name if secret_name else f"{cert_name}-{cert_extension}"
+
+        # iterate over secrets to check if secret with same name exists
+        secret_name = self._check_and_update_secret_name(secrets, secret_name, spc_keyvault_name)
+        self._upload_to_key_vault(secret_name, file, cert_extension)
+
+        # check if there is a spc called "opc-ua-connector", if not create one
+        try:
+            opcua_spc = self.ssc_mgmt_client.azure_key_vault_secret_provider_classes.get(
+                resource_group_name=resource_group,
+                azure_key_vault_secret_provider_class_name=OPCUA_SPC_NAME,
+            )
+        except ResourceNotFoundError:
+            opcua_spc = {}
+
+        self._add_secret_to_spc(
+            secret_name=secret_name,
+            spc=opcua_spc,
+            resource_group=resource_group,
+            spc_keyvault_name=spc_keyvault_name,
+        )
+
+        return self._add_secret_to_secret_sync(
+            secret_name=secret_name,
+            file_name=file_name,
+            secret_sync=opcua_secret_sync,
+            resource_group=resource_group,
+            spc_name=OPCUA_SPC_NAME,
+            secret_sync_name=OPCUA_ISSUER_LIST_SECRET_SYNC_NAME,
+        )
+
+    def _process_fortos_yaml(self, object_text: str, secret_entry: Optional[dict] = None) -> str:
+        if object_text:
+            # TODO: formatting will be removed once fortos service fixes the formatting issue
+            object_text.replace("\n    - |", "\n- |")
```

Comment:
I actually don't think we need this line. Since we're loading the object (which seems to remove formatting anyway), and then dumping again after, I believe the last `.replace` is all we need?

Reply:
yeah you are right!